harborclient/v2/shell.py

```diff
@@ -341,15 +342,96 @@ def do_list(cs, args):
     utils.print_list(data, fields, sortby=args.sortby)
 
 
+@utils.arg(
+    '--repository',
+    metavar='<repository>',
+    help='Name of repository to be deleted.')
+@utils.arg(
+    '--repo-reg',
+    metavar='<repo_reg>',
+    help='expression to list the repositories to be deleted')
+@utils.arg(
+    '--dry-run',
+    '-d',
+    dest='dryrun',
+    action="store_true",
+    help="will only print what would have been removed")
+def do_repository_delete(cs, args):
+    """Delete repository"""
+    if args.repo_reg:
+        # list all the repositories to delete
+        repositories = cs.repositories.list(cs.client.project)
+        for repository in repositories:
+            if re.match(args.repo_reg,repository['name']):
+                if args.dryrun:
+                    print("Would have removed : %s" % repository['name'])
+                else:
+                    cs.repositories.delete_repository(repository['name'])
+                    print("Repository %s deleted" % repository['name'])
+    elif args.repository:
+        # unique repository removal
+        cs.repositories.delete_repository(args.repository)
+        print("Repository %s deleted" % args.repository)
+    else:
+        print("No repository of repository expression given")
+
 @utils.arg('repository', metavar='<repository>', help='Name of repository.')
-def do_list_tags(cs, args):
+def do_tags_list(cs, args):
     """Get tags of a relevant repository."""
     tags = cs.repositories.list_tags(args.repository)
     fields = ["name", 'author', 'architecture',
               'os', 'docker_version', 'created']
     utils.print_list(tags, fields, sortby="name")
 
 
+@utils.arg('repository', metavar='<repository>', help='Name of repository.')
+@utils.arg('tag', metavar='<tag>', help='Name of the tag.')
+def do_tags_delete(cs, args):
+    """delete tag of a relevant repository."""
+    rc = cs.repositories.delete_tags(args.repository, args.tag)
+
+    print("Delete tag '%s:%s' sucessfullywith return code %s" %
```

Comment:
It's much better if you can catch exception such as `404 Not Found`.

Reply:
I'm trying to catch the exception, but it seems that the client doesn't raise any exception, and the body is systematically empty. I'll try to have more details about the return of harbor on the DELETE action, but I can't seem to make it failed...  
If you have any idea ?

Reply:
Okay FYI Harbor doesn't return 404 on a tag not found but a 404 for a repository not found (with a message `project not found`)